src/libPMacc/include/particles/memory/boxes/ParticlesBox.hpp

```diff
@@ -64,7 +67,16 @@ class ParticlesBox
      */
     HDINLINE FRAME &getEmptyFrame()
     {
-        return data.pop();
+        FrameType* tmp = new FrameType;
+
+        if (tmp != NULL)
+        {
+            /* delete all particles we can not assume that new memory is zeroed */
```

Comment:
better:
delete all particles **since** we can not assume that **newly allocated** memory contains **zeros**

Do you realy *delete* particles here or do you set *some attribut* of them to zero?